application/src/main/java/org/opentripplanner/graph_builder/module/osm/VertexGenerator.java

```diff
@@ -55,10 +73,15 @@ public VertexGenerator(
    *
    * @param node The node to fetch a label for.
    * @param way  The way it is connected to (for fetching level information).
+   * @param linearBarrierNodeType How should the node be handled if it is on a linear barrier
    * @return vertex The graph vertex. This is not always an OSM vertex; it can also be a
    * {@link OsmBoardingLocationVertex}
    */
-  IntersectionVertex getVertexForOsmNode(OsmNode node, OsmEntity way) {
+  IntersectionVertex getVertexForOsmNode(
+    OsmNode node,
```

Comment:
I wonder if getVertexForOsmNode should ber overloaded by 2 versions, one of which does not require linearBarrierNodeType. Now the type is visible in places where it is not relevant at all, like WalkableAreaBuilder.

Reply:
WalkableAreaBuilder is the place where it is relevant - it is requesting to split nodes on linear barriers.